Fixing broken test helper function exposes test failures around MAX_PASSWORD

I was curious enough about #1251 that I poked at the auth tests to see why that wasn't already exposed. I did find that `get_max_password_length()` is broken and fixing it exposes two failures:

https://github.com/pgbouncer/pgbouncer/commit/5ca35d49eef7a7738ef00ae137109d13cddfc2bc

```
FAILED test/test_auth.py::test_password_server - psycopg.OperationalError: connection failed: connection to server at "127.0.0.1", port 10209 failed: FATAL: ...
FAILED test/test_auth.py::test_password_client - psycopg.OperationalError: connection failed: connection to server at "127.0.0.1", port 10212 failed: FATAL: ...

```

This test fails also on `main` and the proposed fix for #1251 does not fix this so it appears to be unrelated.

I do, however, wonder if the above isn't related to what was reported [here](https://github.com/pgbouncer/pgbouncer/issues/560) 

I am getting the following in the logs:

```2025-02-08 20:44:34.944 UTC [1652222] WARNING server login failed: FATAL empty password returned by client
2025-02-08 20:44:34.944 UTC [1652222] LOG C-0x55e57ebcfb20: p4l/postgres@127.0.0.1:52034 closing because: empty password returned by client (age=0s)
2025-02-08 20:44:34.944 UTC [1652222] WARNING C-0x55e57ebcfb20: p4l/postgres@127.0.0.1:52034 pooler error: empty password returned by client
2025-02-08 20:44:34.944 UTC [1652222] LOG S-0x55e57ebe3a70: p4l/longpass@127.0.0.1:10201 closing because: empty password returned by client (age=0s)
```

Still digging...

--- a/test/utils.py
+++ b/test/utils.py
@@ -143,9 +143,10 @@ PG_MAJOR_VERSION = get_pg_major_version()
 
 def get_max_password_length():
     with open("../include/bouncer.h", encoding="utf-8") as f:
-        match = re.search(r"#define MAX_PASSWORD\s+([0-9])", f.read())
+        match = re.search(r"#define MAX_PASSWORD\s+([0-9].*)", f.read())
         assert match is not None
         max_password_length = int(match.group(1))
+        assert max_password_length >= 996
 
     if max_password_length > 996 and PG_MAJOR_VERSION < 14:
         return 996
@@ -154,7 +155,7 @@ def get_max_password_length():
 
 PKT_BUF_SIZE = 4096
 MAX_PASSWORD_LENGTH = get_max_password_length()
-LONG_PASSWORD = "a" * MAX_PASSWORD_LENGTH
+LONG_PASSWORD = "a" * (MAX_PASSWORD_LENGTH - 1)
 
 PG_SUPPORTS_SCRAM = PG_MAJOR_VERSION >= 10
 

